### PR TITLE
Multiple changes to archive and emulator classes

### PR DESCRIPTION
**Pull request by andreufont**
Several week long discussion on archives, training sets, emulators, and more. 

We are not yet 100% happy with the situation, but it is good enough for now, and we will come back to this after the DESI meeting

### CODE REVIEW

**Review by jchavesmontero — APPROVED** [review]
Plenty of things left to implement and check, but I would really like to merge this branch :)

**Review by lauracabayol — APPROVED** [review]

**Comment by andreufont** [comment]
I would like to review them a bit more closely next week, but we'll try to merge soon

File: lace/archive/interface_archive.py
Comment (andreufont) [inline lace/archive/interface_archive.py:1]: We should remove this file, isn't it @jchavesmontero ?

File: lace/archive/p1d_archive_Nyx.py
Comment (andreufont) [inline lace/archive/p1d_archive_Nyx.py:1]: We should also delete this file, right @jchavesmontero ?

File: lace/archive/p1d_archive.py
Comment (andreufont) [inline lace/archive/p1d_archive.py:1]: And this one?

File: lace/cosmo/thermal_model.py
Comment (andreufont) [inline lace/cosmo/thermal_model.py:1]: @jchavesmontero - why do we have this thermal model file here? This should be part of cup1d, not LaCE. If you need a specific function (to compute sigmaT_Mpc) you can copy it, but we can not add the whole ThermalModel module to LaCE.

File: lace/emulator/emulator.py
Comment (andreufont) [inline lace/emulator/emulator.py:1]: @lauracabayol - do we still need this file? Can we delete it? If not, can we rename it?

**Comment by andreufont** [comment]
new_cup1d

Yaaaaaay :-)

File: lace/emulator/emulator.py
Comment (lauracabayol) [reply lace/emulator/emulator.py:1]: We can delete it!

File: lace/archive/p1d_archive_Nyx.py
Comment (lauracabayol) [reply lace/archive/p1d_archive_Nyx.py:1]: This is the old file I was using, I think we can delete it as well

File: lace/archive/p1d_archive.py
Comment (andreufont) [reply lace/archive/p1d_archive.py:1]: I've deleted it

File: lace/cosmo/thermal_model.py
Comment (andreufont) [reply lace/cosmo/thermal_model.py:1]: I removed that file, and added instead a very short file with the only relevant functions needed to go from temperatuer to thermal broadening. These functions will then be used from cup1d, to have consistent definitions.

**Comment by andreufont** [comment]
@lauracabayol - I've done several minor changes to address my own comments during my review. Could you check for the last time that I have not broken anything? 

I would then proceed with the merge.

File: lace/archive/interface_archive.py
Comment (jchavesmontero) [reply lace/archive/interface_archive.py:1]: Yes!

**Pull request merged**